### ISSUE TITLE
Region is not resolved from config file when none specified

### ISSUE THREAD

**Issue by petrisorciprian-vitals** [async-aws/aws#1585]
Hello,

Firstly, thank you so much for maintaining such a project, it really is miles beyond what AWS offers in terms of developer experience.
Now, onto why I raised this issue:

I recently ran into a problem where the region used (when not specifying one in code/envs) differs between this SDK and the official AWS SDK.
Namely, the official AWS SDK reads the region as well from the config file (~/.aws/config) while this one seems not to.

I'm not sure how similar this project aims to be to the official AWS SDK in this regard; my expectation was that credentials/regions are defaulting in the same way as the official SDK if none are provided.

Is this the case? 

If not, perhaps a notice somewhere in the documentation (I wasn't able to find one) that this scenario behaves differently might be useful to prevent confusion.

For a minimal reproduction, some steps:
1. Set up a default profile in ~/.aws/config and ~/.aws/credentials:
```
// config file
[default]
region = us-east-2
```

and 

```
// credentials file
[default]
aws_access_key_id = <KEY>
aws_secret_access_key = <SECRET>
```
2. Create a client without specifying anything and make a request:
```php

$client = new CloudWatchClient();
$client->putMetricData(...); // operation doesn't matter
```
3. Observe that us-east-1 is used instead of us-east-2, which is specified in the default profile configuration.

Bonus: Here's where the official SDK loads the region from the config file:
https://github.com/aws/aws-sdk-php/blob/master/src/ClientResolver.php#L1191
which is called from https://github.com/aws/aws-sdk-php/blob/master/src/AwsClient.php#L222.

Cheers!

**Comment by jderusse** [comment]
Hello, thank you for the well-explained report.
Are you sure about the minimal reproducer? I don't have the same result, in my case the region defined in `~/.aws/config` is used.

The current implementation is:
- When instantiating a client, the configuration object is built.
- The configuration merge data from multiple source (env variables, .ini file, ...) https://github.com/async-aws/aws/blob/7feeeb725fe211ab2b7d147bbefcd63993103bff/src/Core/src/Configuration.php#L111-L119
- It load .ini files here: https://github.com/async-aws/aws/blob/7feeeb725fe211ab2b7d147bbefcd63993103bff/src/Core/src/Configuration.php#L217-L237
- multiple ini files are merged together here: https://github.com/async-aws/aws/blob/7feeeb725fe211ab2b7d147bbefcd63993103bff/src/Core/src/Credentials/IniFileLoader.php#L62-L73


As expected, if the region is defined in `~/.aws/config` and credentials are defined in ` ~/.aws/credentials` then both file are merged, and region from `~/.aws/config` will be used.

Could you help to troubleshoot by dumping the configuration and checking if the region is set:
- in `src/Core/src/AbstractApi.php` dump the `$configuration` at line 80
- in `src/Core/src/AbstractApi.php` dump the `$region variable` at line 240

**Comment by petrisorciprian-vitals** [comment]
Hi @jderusse !

Thank you for the hints, I've debugged this and found out that the project I'm working on had the following env set:
```
AWS_CONFIG_FILE=false
```

I now understand why the async-aws SDK behaved the way it did, since my ~/.aws/config file is not loaded.

What I'm not okay with is the fact that the official AWS SDK seemed to be just fine with this set as such.
I tracked this down to the following code on their side:
https://github.com/aws/aws-sdk-php/blob/master/src/AbstractConfigurationProvider.php#L106-L111

The function above calls getenv(self::ENV_CONFIG_FILE), which will return false (assuming AWS_CONFIG_FILE=false in env).
Thus, their if check fails and it goes into the ```return self::getHomeDir() .'/.aws/config';``` below.

Not sure what to think on this one, I find the behaviour of async-aws to be more correct, but this difference is kind of a headache to debug and can potentially be very destructive when migrating from one sdk to another.

I'm curious what your thoughts are on this, I'm currently tempted to raise an issue in the official SDK about this.
